opendevin/llm/llm.py

```diff
@@ -29,6 +31,12 @@ def __init__(self,
 
         completion_unwrapped = self._completion
 
+        def after_each_retry(retry_state):
+            llm_prompt_logger.info(f'Attempt number {retry_state.attempt_number}')
+
+        @retry(reraise=True,
+               stop=stop_after_attempt(num_retries),
+               wait=wait_random_exponential(min=1, max=5), retry=retry_if_not_exception_type(AuthenticationError), after=after_each_retry)
```

Comment:
IMO we want to add a lot more exception types here

https://docs.litellm.ai/docs/exception_mapping

We should only retry on:
* APIConnectionError
* InternalServerError
* RateLimitError

Everything else we should fail

Reply:
This reminds me that 

```python
            if isinstance(e, APIConnectionError):
                time.sleep(3)
```

from `agent_controller.py` should be removed with this PR.

Reply:
Seems, they don't have exception for InternalServerError https://github.com/BerriAI/litellm/blob/main/litellm/exceptions.py